sled-agent/src/serial.rs

```diff
@@ -179,15 +180,20 @@ impl SerialConsoleBuffer {
                         }
                     }
                 });
+            // how long to sleep when nothing happens (to avoid thrashing CPU with polling here)
+            const SLEEP_DURATION: Duration = Duration::from_secs(1);
             match connect_future.await {
                 Ok((mut websocket, _)) => loop {
                     match websocket.next().await {
-                        None => {}
+                        None => {
+                            tokio::time::sleep(SLEEP_DURATION).await;
+                        }
```

Comment:
Since we're relying on the [`tokio_tungstenite::WebSocketStream`'s `Stream`](https://docs.rs/tokio-tungstenite/latest/tokio_tungstenite/struct.WebSocketStream.html#impl-Stream) implementation, I looked under the hood:

https://docs.rs/tokio-tungstenite/latest/src/tokio_tungstenite/lib.rs.html#265-282

It looks like `None` is returned from `next()` when the connection has been closed - what's the point of sleeping in this case?